api/core/v1alpha1/dnset_types.go

```diff
@@ -7,6 +7,14 @@ import (
 type DNSetSpec struct {
 	PodSet `json:",inline"`
 
+	// Volume is the local persistent volume for each dn service isntance
+	// +required
+	Volume Volume `json:"volume"`
+
+	// ShardStorage is an external shard storage shared by all dn service instances
+	// +required
+	SharedStorage SharedStorageProvider `json:"sharedStorage"`
```

Comment:
This should be read from LogSet since using a different `SharedStorage` for DN is an error